libcperciva/datastruct/seqptrmap.c

```diff
@@ -74,7 +79,18 @@ seqptrmap_add(struct seqptrmap * M, void * ptr)
 void *
 seqptrmap_get(struct seqptrmap * M, int64_t i)
 {
-	size_t pos = i - M->offset;
+	size_t pos;
+	int64_t pos_i;
+
+	/* No valid pointer is less than the offset. */
+	if (i < M->offset)
+		return (NULL);
+
+	/* Check that the position will fit into size_t. */
+	pos_i = i - M->offset;
+	if ((uint64_t)pos_i > (uint64_t)SIZE_MAX)
+		return (NULL);
+	pos = (size_t)pos_i;
```

Comment:
Definitely better, but I think this code can be simplified further.  How about checking if
`(uint64_t)(i - M->offset) < (uint64_t)(M->len)` rather than checking if we can cast down to size_t and then checking the size_t value?